### PR TITLE
Added logic for parameters and supertypes lists to NewlinesRule

### LINKED ISSUE #420: Long supertype list on the new line
For classes with a long supertype list, put a line break after the colon and align all supertype names horizontally:

```kotlin​
class MyFavouriteVeryLongClassHolder :
    MyLongHolder<MyFavouriteVeryLongClass>(),
    SomeOtherInterface,
    AndAnotherOne {
​
}
```

### PR DESCRIPTION
**Pull request by petertrr**
### What's done:
* Changed logic
* Added tests
* Updated code style guide

This pull request closes #420

## Actions checklist
* [x] Implemented Rule, added Warnings
* [x] Added tests on checks
* [x] Added tests on fixers
* [x] Updated diktat-analysis.yml

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt [hunk 1]
```diff
@@ -296,6 +306,64 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
...
+            }
+    }
+
+    @Suppress("UnsafeCallOnNullableType")
+    private fun ASTNode.getParentIdentifier() = when (treeParent.elementType) {
+        PRIMARY_CONSTRUCTOR -> treeParent.treeParent
+        SECONDARY_CONSTRUCTOR -> parent(CLASS)!!
+        else -> treeParent
+    }
+        .getIdentifierName()?.text
```
Comment (kentr0w) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt:366]: I found that It returns `null` when there is lambda.

**Review by kentr0w — CHANGES_REQUESTED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt (hunk 1 above)
Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt:366]: Where?

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt:366]: Oh, I see, rule doesn't distinguish functional types from function declarations. Good point, I'll fix it.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt [hunk 2]
```diff
@@ -296,6 +308,69 @@ class NewlinesRule(private val configRules: List<RulesConfig>) : Rule("newlines"
...
+        if (node.elementType == VALUE_PARAMETER_LIST && node.treeParent.elementType.let { it == FUNCTION_TYPE || it == FUNCTION_TYPE_RECEIVER }) {
+            // do not check other value lists
+            return
+        }
+
+        val (numEntries, entryType) = when (node.elementType) {
+            VALUE_PARAMETER_LIST -> (node.psi as KtParameterList).parameters.size to "value parameters"
+            SUPER_TYPE_LIST -> (node.psi as KtSuperTypeList).entries.size to "supertype list entries"
+            else -> {
+                log.warn("Unexpected node element type ${node.elementType}")
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt:325]: I guess we have decided that such warnings can be raised only in critical cases. This not that one :)

Comment (petertrr) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/files/NewlinesRule.kt:325]: It's not critical, but rather highly improbable. I just wanted to make sure that if this branch ever executes, we have data on why we got there.

**Review by akuleshov7 — APPROVED** [review]
lgtm

**Review by kentr0w — APPROVED** [review]
Lgtm

**Pull request merged**